tools/hansel/bio_hansel.xml

```diff
@@ -1,227 +1,283 @@
-                </assert_contents>
-            </output>
-        </test>
-        <test>
-            <param name="type" value="paired"/>
-            <param name="type_of_scheme" value="heidelberg"/>
-            <param name="fastq_input1" value="SRR5646583_SMALL_1.fastq"/>
-            <param name="fastq_input2" value="SRR5646583_SMALL_2.fastq"/>
-            <output name="results.tab">
-                <assert_contents>
-                    <has_text_matching expression="sample\s+scheme\s+scheme_version\s+subtype\s+all_subtypes\s+tiles_matching_subtype\s+are_subtypes_consistent\s+inconsistent_subtypes\s+n_tiles_matching_all\s+n_tiles_matching_all_expected\s+n_tiles_matching_positive\s+n_tiles_matching_positive_expected\s+n_tiles_matching_subtype\s+n_tiles_matching_subtype_expected\s+file_path\s+avg_tile_coverage\s+qc_status\s+qc_message"/>
-                </assert_contents>
-            </output>
-            <output name="match_results.tab">
-                <assert_contents>
-                    <has_text_matching expression="tilename\s+seq\s+freq\s+refposition\s+subtype\s+is_pos_tile\s+is_kmer_freq_okay\s+sample\s+scheme\s+scheme_version\s+qc_status\s+qc_message"/>
-                </assert_contents>
-            </output>
-            <output name="tech_results.tab">
-                <assert_contents>
-                    <has_text_matching expression="sample\s+subtype\s+avg_tile_coverage\s+qc_status\s+qc_message"/>
-                </assert_contents>
-            </output>
-        </test>
-    </tests>
-    <help><![CDATA[
+<tool id="bio_hansel" name="bio_hansel" version="2.0.0">
+  <description>SNV subtyping of genome sequence reads or assemblies</description>
+  <requirements>
+    <requirement type="package" version="2.0.0">bio_hansel</requirement>
+  </requirements>
+  <command detect_errors="exit_code">
+<![CDATA[
+
+#def is_gzipped_fastq($data_input)
+  ## Is FASTQ data param gzipped type? i.e. either 'fastq.gz' or 'fastqsanger.gz'?
+  #return $data_input.is_of_type('fastqsanger.gz') or $data_input.is_of_type('fastq.gz')
+#end def
+
+#def get_fastq_ext($data_input)
+  ## Get file extension for FASTQ data param
+  #return '.fastq.gz' if $is_gzipped_fastq($data_input) else '.fastq'
+#end def
+
+#def get_paired_fastq_filename($data_input, $name=None, $is_forward=True)
+  ## Get paired FASTQ filename for a data param with appropriate file extension 
+  ## with '_1' or '_2' appended if forward or reverse reads, respectively.
+  #set $name = $name if $name is not None else $data_input.name
+  #set $postfix = '1' if $is_forward else '2'
+  #set $ending = '_{}{}'.format($postfix, $get_fastq_ext($data_input))
+  #return '"{}"'.format($name) if $ending in $name else '"{}{}"'.format($name, $ending)
+#end def
+
+## Create symlinks from Galaxy *.dat to <sample_name>(.fasta|.fastq|.fastq.gz)
+#if $input.type == 'fasta'
+#set $input_files = '"{}"'.format($input.fasta.name)
+  ln -s "$input.fasta" $input_files &&
+#elif $input.type == 'paired'
```

Comment:
I found that when I select the **paired** input type (instead of **paired_collection**), `bio_hansel` appears to be treating the pairs as two separate single-end files.

For example, if I use the following as input:

![image](https://user-images.githubusercontent.com/200517/39816263-5af9bf62-5360-11e8-99cc-0e4c8e4700f6.png)

I get the following as output:

![image](https://user-images.githubusercontent.com/200517/39816219-3e79963c-5360-11e8-8cea-db91244ea03a.png)

However, if I pass the same files stored in a paired-end collection, I get a single entry in the result:

![image](https://user-images.githubusercontent.com/200517/39816324-8810365c-5360-11e8-85ff-56046322b51e.png)

I'm wondering if this has something to do with the way `bio_hansel` is reading file names to determine if they are paired-end or not?

Reply:
Yes, `bio_hansel` is expecting reads belonging to the same sample to have the same base sample name encoded in the filename followed by `_\d+` and the file extension.  It expects `<sample/run_name>_(1|2).fastq(.gz)?` which is seen with data retrieved from public databases like ENA or SRA (after `fastq-dump --split-files ...`).

Reply:
Okay, thanks.  Could you make an issue for `bio_hansel` to also add the regex `_(R1|R2)` since that's how the files are output from the Illumina instruments.  I will merge this code in once you do that (and additional regex can be for another release).

Reply:
A regex replacement in the Galaxy tool would be great and should work out :+1:

Reply:
@apetkau Added better handling of `_R(1|2)` in filenames of input files in 411bacd11d95c2bff653992c9f4c98ee8863e015. Running `bio_hansel` with the above input with R1/R2 with `Paired-end FASTQs` input type should result in only one result entry.

Reply:
Thanks Peter, it's working now :+1: